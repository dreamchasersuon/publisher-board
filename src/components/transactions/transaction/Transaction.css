.Transaction {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
  padding-bottom: 15px;
  border-bottom: 1px solid #e5e5e5;
}

.Transaction:last-child {
  border: none;
  margin-bottom: 0;
  padding-bottom: 0;
}

.Transaction-date {
  margin-right: auto;
  font-size: 14px;
  margin-bottom: 10px;
}

.Transaction-comment_and_amount {
  display: flex;
  justify-content: space-between;
}

.Transaction-comment {
  color: #666;
}

.Transaction-total {
  font-size: 14px;
}
