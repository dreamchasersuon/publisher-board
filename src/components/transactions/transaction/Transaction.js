import React from 'react';
import PropTypes from 'prop-types';
import VirtualItem from './virtual-item/VirtualItem';
import './Transaction.css';

const months = {
  1: 'Jan',
  2: 'Feb',
  3: 'Mar',
  4: 'Apr',
  5: 'May',
  6: 'Jun',
  7: 'Jul',
  8: 'Aug',
  9: 'Sep',
  10: 'Oct',
  11: 'Nov',
  12: 'Dec',
};

export default function Transaction({ transaction }) {
  function formatDate(dateUTC) {
    const date = new Date(dateUTC);
    return `${date.getDate()} ${months[date.getMonth()]} ${date.getFullYear()}`;
  }
  return (
    <div className="Transaction">
      <span className="Transaction-date">{formatDate(transaction.date)}</span>
      {transaction.virtual_items &&
        transaction.virtual_items.map(item => {
          return <VirtualItem key={item.virtual_item.sku} item={item} />;
        })}
      <div className="Transaction-comment_and_amount">
        {<span className="Transaction-comment">{transaction.comment}</span>}
        <span className="Transaction-total">
          {'Total: '}
          {transaction.virtual_items
            ? transaction.virtual_items.reduce(
                (sum, item) => sum + (item.amount || 0),
                0,
              )
            : transaction.amount}
        </span>
      </div>
    </div>
  );
}

Transaction.propTypes = {
  transaction: PropTypes.object,
};
