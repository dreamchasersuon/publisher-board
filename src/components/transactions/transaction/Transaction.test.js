import React from 'react';
import snapshotDiff from 'snapshot-diff';
import Transaction from './Transaction';

const transactionWithItems = {
  amount: 12,
  comment: null,
  date: '2018-12-28T15:02:29+03:00',
  operation_id: 178199125,
  virtual_items: [
    {
      amount: 1,
      virtual_item: {
        localized_name: 'Item2',
        sku: '002',
      },
    },
    {
      amount: 123,
      virtual_item: {
        localized_name: 'Item3',
        sku: '003',
      },
    },
  ],
};

const transactionWithComment = {
  amount: 1200,
  comment: 'Update balance',
  date: '2017-02-10T15:02:29+03:00',
  operation_id: 828328435,
};

test('renders purchase transaction or update balance', () => {
  expect(
    snapshotDiff(
      <Transaction transaction={transactionWithItems} />,
      <Transaction transaction={transactionWithComment} />,
    ),
  ).toMatchSnapshot();
});
