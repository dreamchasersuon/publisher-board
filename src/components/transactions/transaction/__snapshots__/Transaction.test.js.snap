// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders purchase transaction or update balance 1`] = `
"Snapshot Diff:
- <Transaction transaction={{\\"amount\\": 12, \\"comment\\": null, \\"date\\": \\"2018-12-28T15:02:29+03:00\\", \\"operation_id\\": 178199125, \\"virtual_items\\": [{\\"amount\\": 1, \\"virtual_item\\": {\\"localized_name\\": \\"Item2\\", \\"sku\\": \\"002\\"}}, {\\"amount\\": 123, \\"virtual_item\\": {\\"localized_name\\": \\"Item3\\", \\"sku\\": \\"003\\"}}]}} />
+ <Transaction transaction={{\\"amount\\": 1200, \\"comment\\": \\"Update balance\\", \\"date\\": \\"2017-02-10T15:02:29+03:00\\", \\"operation_id\\": 828328435}} />

@@ -2,49 +2,23 @@
    className=\\"Transaction\\"
  >
    <span
      className=\\"Transaction-date\\"
    >
-     28 Nov 2018
+     10 Jan 2017
    </span>
    <div
-     className=\\"Transaction-item\\"
+     className=\\"Transaction-comment_and_amount\\"
    >
      <span
-       className=\\"TransactionItem-name\\"
+       className=\\"Transaction-comment\\"
      >
-       Item2
-     </span>
-     <span
-       className=\\"TransactionItem-amount\\"
-     >
-       1
-     </span>
-   </div>
-   <div
-     className=\\"Transaction-item\\"
-   >
-     <span
-       className=\\"TransactionItem-name\\"
-     >
-       Item3
+       Update balance
      </span>
      <span
-       className=\\"TransactionItem-amount\\"
-     >
-       123
-     </span>
-   </div>
-   <div
-     className=\\"Transaction-comment_and_amount\\"
-   >
-     <span
-       className=\\"Transaction-comment\\"
-     />
-     <span
        className=\\"Transaction-total\\"
      >
        Total: 
-       124
+       1200
      </span>
    </div>
  </div>"
`;
